src/flint/fmpz_factor.jl

```diff
@@ -18,30 +18,45 @@ function _factor(a::ZZRingElem)
 end
 
 # Just to give a helpful error message if someone tries to factor a boolean
-function factor(b::Bool)
-  throw(DomainError("Cannot factorize a boolean"));
+function factor(::Bool)
+  throw(DomainError("Cannot factorize a boolean"))
 end
 
-# This function handles machine integer types (up to 64 bits)
-function factor(a::T) where T <: Union{Int, UInt}
+function factor(::Type{Bool}, ::Int)
+  throw(DomainError("Cannot have a factorization into booleans"))
+end
+
+function factor(::Type{Bool}, ::ZZRingElem)
+  throw(DomainError("Cannot have a factorization into booleans"))
+end
```

Comment:
```suggestion
function factor(::Type{Bool}, ::IntegerUnion)
  throw(DomainError("Cannot have a factorization into booleans"))
end
```
we can condense these two into one

Reply:
Did not work: Julia reports ambiguity:
```
julia> factor(Bool, 1)
ERROR: MethodError: factor(::Type{Bool}, ::Int64) is ambiguous.

Candidates:
  factor(::Type{Bool}, ::Union{Integer, ZZRingElem})
    @ Nemo ~/OSCAR/GIT/Nemo.jl/src/flint/fmpz_factor.jl:25
  factor(::Type{T}, a::Int64) where T<:Integer
    @ Nemo ~/OSCAR/GIT/Nemo.jl/src/flint/fmpz_factor.jl:38

Possible fix, define
  factor(::Type{Bool}, ::Int64)
```
Similar problem if I use `::Integer`.  Is there a clean way of resolving this?